File: src/main/kotlin/com/wynnlab/spells/assassin/AssassinMain.kt
package com.wynnlab.spells.assassin

/*import com.wynnlab.spells.SpellData
import org.bukkit.Particle
import org.bukkit.Sound
import org.bukkit.entity.Mob
import org.bukkit.entity.Player

class AssassinMain(player: Player) : SpellL(player, 1, SpellData.METEOR) {
    override fun tick() {
        player.playSound(player.location, Sound.ENTITY_PLAYER_ATTACK_SWEEP, 1f, .8f)
        val l = player.eyeLocation + player.eyeLocation.direction * 2.0
        player.spawnParticle(Particle.SWEEP_ATTACK, l, 1, .0, .0, .0, .0)

        for (e in player.world.getNearbyEntities(l, 2.0, 2.0, 2.0)) {
            if (e is Player)
                continue
            if (e !is Mob)
                continue
            e.damage(2.0, player)
            e.noDamageTicks = 0
        }
    }
}*/

File: src/main/resources/classes/assassin/assassin_main.py
#function tick() {
#    this.sound(Sound.ENTITY_PLAYER_ATTACK_SWEEP, 1, .8);
#    var l = this.player.getEyeLocation().clone().add(this.player.getEyeLocation().getDirection());
#    this.particle(l, Particle.SWEEP_ATTACK, 1, 0, 0, 0, 0);
#
#    this.nearbyMobs(l, 2, 2, 2).forEach(function (e) {
#        this.damage(e, 2);
#    });
#}
from org.bukkit import Particle, Sound
from org.bukkit.potion import PotionEffectType

from com.wynnlab.spells import PySpell

class Spell(PySpell):
    def tick(self):
        if self.player.hasPotionEffect(PotionEffectType.INVISIBILITY):
            self.castSpell('ASSASSIN', 5)

        self.sound(Sound.ENTITY_PLAYER_ATTACK_SWEEP, 1, .8)
        l = self.player.getEyeLocation().clone().add(self.player.getEyeLocation().getDirection())
        self.particle(l, Particle.SWEEP_ATTACK, 1, 0, 0, 0, 0)

        for e in self.nearbyMobs(l, 2, 2, 2):
            self.damage(e, 2)